pkg/consolelink/main.go

```diff
@@ -0,0 +1,45 @@
+package consolelink
+
+import (
+	"fmt"
+
+	consolev1 "github.com/openshift/api/console/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+)
+
+// RouteAnnotation used to annotate the link with the route name
+var RouteAnnotation = "consolelink.jaegertracing.io/route"
+
+// Get returns an ConsoleLink specification for the current instance
+func Get(jaeger *v1.Jaeger, route *routev1.Route) *consolev1.ConsoleLink {
+	return &consolev1.ConsoleLink{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: jaeger.Namespace + ".jaeger-consolelink-" + jaeger.Name,
+			Namespace:jaeger.Namespace,
+			Labels: map[string]string{
+				"app.kubernetes.io/instance":   jaeger.Name,
+				"app.kubernetes.io/managed-by": "jaeger-operator",
+			},
+			Annotations: map[string]string{
+				RouteAnnotation: route.Name,
+			},
+		},
+		Spec: consolev1.ConsoleLinkSpec{
+			Location:        consolev1.ApplicationMenu,
+			ApplicationMenu: &consolev1.ApplicationMenuSpec{},
+			Link: consolev1.Link{
+				Text: "Jaeger [" + jaeger.Name + "." + jaeger.Namespace + "]",
```

Comment:
Can you remove the namespace from the link text, as it will only be shown when viewing that namespace in the console.

Reply:
Sure

Reply:
Done